ext/web/06_streams.js

```diff
@@ -4379,7 +4380,7 @@
    * @returns {IteratorResult<T>}
    */
   function createIteratorResult(value, done) {
-    const result = ObjectCreate(null);
+    const result = ObjectCreate(ObjectPrototype);
```

Comment:
See: https://github.com/web-platform-tests/wpt/blob/0d21c20b40df4f3f1cb6418a2eedd494257543bb/streams/readable-streams/async-iterator.any.js#L12

Reply:
Is it thus necessary to create this with `ObjectCreate(ObjectPrototype)` or would the simple `{}` be enough? Or does that risk prototype pollution as well?

Reply:
It is needed, after `Object.defineProperties` is called `Object.getPrototypeOf(result) === Object.prototype;` is no longer true, and that assertion fails.

https://github.com/denoland/deno/blob/1e7c3c94c4fa3d38232ee9737b1b281370e30909/ext/web/06_streams.js#L4382-L4394

Reply:
What about `return { value, done }`?

Reply:
@aduh95 that was my first thought, but I wanted to make minimal changes only to fix the tests, both yield the same results. TBH I would remove `createIteratorResult` to avoid the extra overhead.

Reply:
> TBH I would remove createIteratorResult to avoid the extra overhead.

+1, it's only used 5 times in the whole codebase, so it wouldn't be a lot of work indeed.

Reply:
This is correct as per. https://tc39.es/ecma262/#sec-createiterresultobject